tests/test_other.py

```diff
@@ -8311,16 +8311,28 @@ def test_separate_dwarf(self):
     self.assertNotIn(b'subdir/output.wasm.debug.wasm', wasm)
     self.assertNotIn(bytes(os.path.join('subdir', 'output.wasm.debug.wasm'), 'ascii'), wasm)
 
-    # Check that the dwarf file has only dwarf and name sections
+    # Check that the dwarf file has only dwarf, name, and non-code sections
     debug_wasm = webassembly.Module('subdir/output.wasm.debug.wasm')
     if not debug_wasm.has_name_section():
       self.fail('name section not found in separate dwarf file')
     for sec in debug_wasm.sections():
-      if sec.type != webassembly.SecType.CUSTOM:
-        self.fail(f'non-custom section type {sec.type} found in separate dwarf file')
-      elif sec.name != 'name' and not sec.name.startswith('.debug'):
+      # TODO: check for absence of code section (see
+      # https://github.com/emscripten-core/emscripten/issues/13084)
+      if sec.name and sec.name != 'name' and not sec.name.startswith('.debug'):
         self.fail(f'non-debug section "{sec.name}" found in separate dwarf file')
 
+    # Check that dwarfdump can dump the debug info
+    dwdump = self.run_process(
+      [LLVM_DWARFDUMP, 'subdir/output.wasm.debug.wasm', '-name', 'main'],
+      stdout=PIPE).stdout
+    # Basic check that the debug info is more than a skeleton. If so it will
+    # have a subprogram descriptor for main
+    self.assertIn('DW_TAG_subprogram', dwdump)
+    self.assertIn('DW_AT_name\t("main")', dwdump)
+
+    self.run_process([LLVM_DWP, '-e', 'subdir/output.wasm.debug.wasm',
+                      '-o', 'subdir/output.wasm.debug.wasm.dwp'])
```

Comment:
are there any checks we should do on the output file here? Or is this command just run to check for not crashing? (if so do we need the `-o`?)

Reply:
Ah, it was originally to check for not crashing (which is the current symptom) but probably it makes sense to run dwarfdump over the output too.

... And I just realized dwp really only makes sense with -gsplit-dwarf, which we don't have an end-to-end emscripten test for. So I added that.